sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStore.java

```diff
@@ -48,33 +52,40 @@ public class JedisRedisCheckpointStore implements CheckpointStore {
     public Flux<PartitionOwnership> claimOwnership(List<PartitionOwnership> requestedPartitionOwnerships) {
 
         return Flux.fromIterable(requestedPartitionOwnerships).handle(((partitionOwnership, sink) -> {
-
             String partitionId = partitionOwnership.getPartitionId();
             byte[] key = keyBuilder(partitionOwnership.getFullyQualifiedNamespace(), partitionOwnership.getEventHubName(), partitionOwnership.getConsumerGroup(), partitionId);
 
             try (Jedis jedis = jedisPool.getResource()) {
+
                 List<byte[]> keyInformation = jedis.hmget(key, PARTITION_OWNERSHIP);
                 byte[] currentPartitionOwnership = keyInformation.get(0);
 
                 if (currentPartitionOwnership == null) {
                     // if PARTITION_OWNERSHIP field does not exist for member we will get a null, and we must add the field
+                    Long lastModifiedTimeSeconds = Long.parseLong(jedis.time().get(0));
+                    partitionOwnership.setLastModifiedTime(lastModifiedTimeSeconds);
                     jedis.hset(key, PARTITION_OWNERSHIP, DEFAULT_SERIALIZER.serializeToBytes(partitionOwnership));
+                    sink.next(partitionOwnership);
+                    sink.complete();
                 } else {
                     // otherwise we have to change the ownership and "watch" the transaction
                     jedis.watch(key);
-
+                    Long lastModifiedTimeSeconds = Long.parseLong(jedis.time().get(0)) - jedis.objectIdletime(key);
+                    partitionOwnership.setLastModifiedTime(lastModifiedTimeSeconds);
+                    partitionOwnership.setETag("default eTag");
                     Transaction transaction = jedis.multi();
                     transaction.hset(key, PARTITION_OWNERSHIP, DEFAULT_SERIALIZER.serializeToBytes(partitionOwnership));
                     List<Object> executionResponse = transaction.exec();
 
                     if (executionResponse == null) {
                         //This means that the transaction did not execute, which implies that another client has changed the ownership during this transaction
-                        sink.error(new RuntimeException("Ownership records were changed by another client"));
+                        LOGGER.verbose("UNABLE TO CLAIM PARTITION");
```

Comment:
Doesn't need to be in caps... also, which partition were you unable to claim? The message as-is doesn't seem helpful.